rust/src/delta_datafusion.rs

```diff
@@ -367,27 +383,86 @@ impl TableProvider for DeltaTable {
                 .cloned()
                 .collect(),
         ));
-        ParquetFormat::default()
+        let parquet_scan = ParquetFormat::default()
             .create_physical_plan(
                 FileScanConfig {
-                    object_store_url,
+                    object_store_url: self.storage.object_store_url(),
                     file_schema,
                     file_groups: file_groups.into_values().collect(),
                     statistics: self.datafusion_table_statistics(),
                     projection: projection.clone(),
                     limit,
                     table_partition_cols,
+                    config_options: Arc::new(Default::default()),
                 },
                 filters,
             )
-            .await
+            .await?;
+        let delta_scan = DeltaScan {
+            url: self.table_uri(),
+            parquet_scan: parquet_scan,
+        };
+
+        Ok(Arc::new(delta_scan))
     }
 
     fn as_any(&self) -> &dyn Any {
         self
     }
 }
 
+/// A wrapper for parquet scans that installs the required ObjectStore
+#[derive(Debug)]
+pub struct DeltaScan {
+    /// The URL of the ObjectStore root
+    pub url: String,
+    /// The parquet scan to wrap
+    pub parquet_scan: Arc<dyn ExecutionPlan>,
+}
+
+impl ExecutionPlan for DeltaScan {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.parquet_scan.schema()
+    }
+
+    fn output_partitioning(&self) -> Partitioning {
+        self.parquet_scan.output_partitioning()
+    }
+
+    fn output_ordering(&self) -> Option<&[PhysicalSortExpr]> {
+        self.parquet_scan.output_ordering()
+    }
+
+    fn children(&self) -> Vec<Arc<dyn ExecutionPlan>> {
+        vec![self.parquet_scan.clone()]
+    }
+
+    fn with_new_children(
+        self: Arc<Self>,
+        children: Vec<Arc<dyn ExecutionPlan>>,
+    ) -> DataFusionResult<Arc<dyn ExecutionPlan>> {
+        ExecutionPlan::with_new_children(self.parquet_scan.clone(), children)
+    }
```

Comment:
I may be completely off, but can a scan operation even have children? `ParquetScan` itself just returns `Ok(self)` .

Reply:
> can a scan operation even have children?

I would think not, but the goal of this implementation was merely to proxy every call to the wrapped `ParquetScan` with the one additional behavior of registering the `ObjectStore` prior to proxying the `execute()` request... so I just followed that pattern here. I think it's probably best to leave it this way just in case there is some future change where a `ParquetScan` might have children for some reason - possibly because someone wants to register their own proxy in the same way I just did :)